blockchain/types/account/account_serializer.go

```diff
@@ -127,3 +146,47 @@ func (ser *AccountSerializer) UnmarshalJSON(b []byte) error {
 
 	return json.Unmarshal(dec.Account, ser.account)
 }
+
+// UnextendSerializedAccount unextends ExtHash fields within an RLP-encoded account.
+// If the supplied bytes is not an RLP-encoded account, or does not contain any ExtHash,
+// then return the supplied bytes unchanged.
+func UnextendSerializedAccount(b []byte) (result []byte) {
+	acc := safeDecodeRLP(b)
+	if acc == nil {
+		return b // not an account
+	}
+
+	pa := GetProgramAccount(acc)
+	if pa == nil {
+		return b // not a ProgramAccount
+	}
+
+	enc := NewAccountSerializerWithAccount(pa)
+	result, err := rlp.EncodeToBytes(enc)
+	if err != nil {
+		logger.Crit("failed to unextend account blob", "bytes", hexutil.Encode(b), "err", err)
+	}
+	return result
+}
+
+// Account RLP decoder that does not panic
+func safeDecodeRLP(b []byte) Account {
+	if len(b) == 0 {
+		return nil // No type byte
+	}
+
+	var accType AccountType
+	if err := rlp.DecodeBytes(b[0:1], &accType); err != nil {
```

Comment:
How about relocate the blcok 182-184 to here?
`if err := rlp.DecodeBytes(b[:1], &accType); err != nil || accType == LegacyAccountType {`

Reply:
If it's a cosmetic suggestion, can I leave it as it is? I prefer code lines not too long.

Reply:
Sure.